core/trino-main/src/test/java/io/trino/sql/query/TestExpressions.java

```diff
@@ -51,6 +51,12 @@ public void testBooleanExpressionInCase()
         assertThat(assertions.query("VALUES CASE 1 > 2 WHEN true THEN 10 ELSE 20 END")).matches("VALUES 20");
     }
 
+    @Test
+    public void testInShortCircuit()
+    {
+        assertThat(assertions.query("SELECT 3 IN (2, 4, 3, 5 / 0)")).matches("VALUES true");
```

Comment:
I made a change in `ExpressionInterpreter` so that now it doesn't optimize in-predicates by short-circuit.
However, this test still passes, because the generated code for in-predicate does short-circuit. I added comments.